### PR TITLE
dateMultipleFormat - Update1

### PR DESCRIPTION
**Pull request by ltdeta**
- for IE8 compatibility changed `.indexOf` to `$.inArray`
- dateFormater suport for YY
- dateFormater (at the end) causes an error if format-array used

### CODE REVIEW

**Comment by ltdeta** [comment]
can you please check the dateFormater support for YY
my changes are not working correctly

format: ["DD.MM.YYYY","D.M.YY"], 
separator: "." , 
minDate: "21.01.1900", 
outputFormat: "DD.MM.YYYY"

at the moment 1.1.14     is not valid
format["DD.MM.YYYY", "D.M.YY"]

**Comment by Arkni** [comment]
Hi @ltdeta 

the test fail because the `Date` constructor accept dates > `1970`
From [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date):
>The JavaScript date is based on a time value that is milliseconds since midnight 01 January, 1970 UTC. A day holds 86,400,000 milliseconds. The JavaScript Date object range is -100,000,000 days to 100,000,000 days relative to 01 January, 1970 UTC.

So the `minDate` should be > `1970-01-01`

File: src/additional/dateMultipleFormat.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
 	 */
 	function dateHelper( year, month, day ) {
 		// Check for the length of year, month and day
-		if ( year.length !== 4 || day.length > 2 || month.length > 2 || !year || !month || !day ) {
+		if ( (year.length !== 4 && year.length !== 2) || day.length > 2 || month.length > 2 || !year || !month || !day ) {
```
Comment (Arkni) [inline src/additional/dateMultipleFormat.js:16]: You don't need to check for `year.length !== 2`

File: src/additional/dateMultipleFormat.js [hunk 2]
```diff
@@ -25,10 +25,21 @@
...
-		// Check that month is between 1 & 12
-		if ( year < 1000 || year > 9999 || month < 1 || month > 12 ) {
+		// Validate only years between 1000 & 9999 if year.length===4
+		if (year.length===4) {
+		  if ( year < 1000 || year > 9999 ) {
+			return false;
+		  }
+		}
+		// Validate only years between 1 & 99 if year.length===2
+		if (year.length===2) {
```
Comment (Arkni) [inline src/additional/dateMultipleFormat.js:35]: This as well

File: src/additional/dateMultipleFormat.js [hunk 3]
```diff
@@ -60,10 +71,10 @@
 		dateFormat = dateFormat.split( separator );
 
 		var now = new Date(),
-			year  = value[ dateFormat.indexOf( "YYYY" ) ] ||
-			( ( Math.ceil( now.getFullYear() / 100 ) - 1 ) + value[ dateFormat.indexOf( "YY" ) ] ),
-			month = value[ dateFormat.indexOf( "MM" ) ] || value[ dateFormat.indexOf( "M" ) ],
-			day   = value[ dateFormat.indexOf( "DD" ) ] || value[ dateFormat.indexOf( "D" ) ];
+			year        = value[ $.inArray("YYYY",dateFormat) ] ||
+				( ( Math.ceil( now.getFullYear() / 100 ) - 1 ) + value[ $.inArray("YY",dateFormat) ] );
```
Comment (Arkni) [inline src/additional/dateMultipleFormat.js:75]: If year is just 2 characters, for example `15`
So this code 

``` js
( Math.ceil( now.getFullYear() / 100 ) - 1 ) + value[ $.inArray("YY",dateFormat) ]
```

will return `2015` and it's a string
This string will be parsed to `Integer` in `parse` & `dateHelper` methods

File: src/additional/dateMultipleFormat.js [hunk 4]
```diff
@@ -262,11 +273,10 @@
...
 			}
-		}
-
-		// Format the value to match the options.outputFormat param
-		if ( valid && options.outputFormat ) {
-			$( element ).val( dateFormater( parse( value, format, separator ), options.outputFormat ) );
+  		// Format the value to match the options.outputFormat param 
+  		if ( valid && options.outputFormat ) {
+  			$( element ).val( dateFormater( parse( value, format, options.separator ), options.outputFormat ) );
+  		}
```
Comment (Arkni) [inline src/additional/dateMultipleFormat.js:279]: Good catch
I forgot to put this condition inside the `else` block

**Comment by Arkni** [comment]
**NEVER MIND MY PREVIOUS COMMENT**

**Comment by Arkni** [comment]
@ltdeta I will push my change based on yours soon

**Comment by Arkni** [comment]
See commit a8eca8c590577a36cb2fa1a189075bae3a8d58ec

**Comment by ltdeta** [comment]
sql server min date is "01.01.1753"
For that case i can't use this method

**Comment by Arkni** [comment]
@ltdeta 
Did you tested my new change ( commit a8eca8c590577a36cb2fa1a189075bae3a8d58ec )  ?

**Comment by ltdeta** [comment]
now it works , thanks
i have another request :)
an option  for automatically decrease the century if the year is in the future 
01.01.15  => 01.01.2015 valid
01.01.53  => 01.01.2053 invalid because the date is in the future (2015) auto formated  to  01.01.1953

**Comment by Arkni** [comment]
So you want to evaluate the value like that:

```
00 --> 15 => 2000 --> 2015
16 --> 99 => 1916 --> 1999
```

Right ?

**Comment by Arkni** [comment]
here is an implementation for this feature:

``` js
    /**
     * Get the full year from a the last 2 digits of a year
     *      If value > the last 2 digits of the current year
     *      Then return ( 1900 + value )
     *      Else return ( 2000 + value )
     *
     * For example:
     *      value = 75 then the returned value will be '1975'
     *      value = 14 < 15 ( the last 2 digits of the current year) then return 2014
     *
     * @param value The 2 digits year
     *
     * @returns {number}
     */
    function getYear( value ) {
        var date = new Date( parseInt( value, 10 ), 0 ).getFullYear(),
            currentYear  = new Date().getFullYear();

        if ( ( date + 100 ) <= currentYear ) {
            date += 100;
        }

        return date.toString();
    }
```

See the new commit 36c962f00fbcdbbede915367d3d04f87db32029e

**Comment by ltdeta** [comment]
it works fine!
i added the functionality of `getYear()` as a new option for more flexibility

**Comment by ltdeta** [comment]
is it possible extend dateMultipleFormat with `time options`
for a compatible with [jQuery-Timepicker-Addon](https://github.com/trentrichardson/jQuery-Timepicker-Addon)

**Comment by Arkni** [comment]
Yes, it's possible to do that. I will add it when I have time to.

**PS:**

> I forgot to mention that I created a repository for this validator as per request of _**jzaefferer**_, check https://github.com/Arkni/date-validator, and also I removed the `outputFormatOption` as it causes a problem when used whit format like `D{separator}M{separator}YY`.
> Fell free to open an issue in the new repo.

**Pull request closed without merging**